File: api_nestjs_distributed_order_platform/src/application/use-case/order/list-orders.use-case.ts
import { Inject, Injectable } from "@nestjs/common";
import type { OrderRepositoryInterface } from "src/application/port/order.repository";
import { OrderEntity } from "src/domain/entities/order.entity";
import { ORDER_REPOSITORY } from "src/shared/tokens_nest/order.token";

@Injectable()
export class ListOrdersUseCase {
    constructor(
        @Inject(ORDER_REPOSITORY)
        private readonly orderRepository: OrderRepositoryInterface,
    ) {}

    async execute(): Promise<OrderEntity[]> {
        return this.orderRepository.findAll();
    }
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/usecase/order/ListOrdersUseCase.java
package com.project.order.application.usecase.order;

import com.project.order.application.port.OrderRepository;
import com.project.order.domain.model.Order;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ListOrdersUseCase {

    private final OrderRepository orderRepository;

    public ListOrdersUseCase(OrderRepository orderRepository) {
        this.orderRepository = orderRepository;
    }

    public List<Order> execute() {
        return orderRepository.findAll();
    }
}
